src/proxy/health.rs

```diff
@@ -39,6 +39,7 @@ impl Health {
         let default_hook = panic::take_hook();
         panic::set_hook(Box::new(move |panic_info| {
             error!(log, "Panic has occurred. Moving to Unhealthy");
+            error!(log, "Info: {}", panic_info);
```

Comment:
we should probably add the panic info to the first error message rather than log a new one?

Reply:
As a request: If there are small refactorings or additions to a PR that aren't specifically tied to the overall goal of the PR, it would be nice to either split that out into its own small PR we can do a quick review on, or throw a comment on it when submitting the PR / make a note of it in the description so that it's easy to spot.

Going through the code, I didn't even notice that this PR edited `health.rs`, I was just focusing on things that touched on Endpoints, so I totally missed the change.

(Nice catch @iffyio)

Reply:
Yeah I didn't even intend to include this, we have an issue where some tests when they panic they hang, and I needed to add this in order to be able figure out what was panicking, after I fixed the issue I forgot about it.